src/lib/core/selection/pseudo-checkbox/_pseudo-checkbox-theme.scss

```diff
@@ -24,15 +24,15 @@
   }
 
   .mat-pseudo-checkbox-checked, .mat-pseudo-checkbox-indeterminate {
-    &.mat-primary {
+    &.mat-primary, .mat-primary & {
```

Comment:
As we discovered in #4077, adding the `&` of the selector doesn't work with secondary themes